### PR TITLE
updates project setup

### PR DESCRIPTION
**Pull request by brennanMKE**
* adds MIT LICENSE
* adds Build.xcconfig for cross-platform development
* adds workspace to better integrate with Swift packages
* defines Template Macros for source file header
* updates existing source files to use common header

### CODE REVIEW

**Comment by jordibruin** [comment]
Thanks for this @brennanMKE !

**Comment by jordibruin** [comment]
@brennanMKE can you fix the merge conflict? I don't really know what's going wrong 😜

**Comment by jordibruin** [comment]
Also, I noticed that you left your development team in your commit.

Also, I think Swift_Charts_ExamplesApp is better than ChartsApp, since it's a clearer explanation of what the project is.

File: Swift Charts Examples/Charts/LineCharts/Simple/LineChartSimple.swift [hunk 1]
```diff
@@ -1,9 +1,6 @@
 //
-//  LineChartSimple.swift
-//  Swift Charts Examples
-//
-//  Created by Jordi Bruin on 12/06/2022.
-//
+// Copyright © 2022 Swift Charts Examples.
```
Comment (jordibruin) [inline Swift Charts Examples/Charts/LineCharts/Simple/LineChartSimple.swift:2]: Is there a reason to not show the original author here? We're not trying to make a transparant project, and I'd rather highlight the developers who helped contribute.

Comment (brennanMKE) [reply Swift Charts Examples/Charts/LineCharts/Simple/LineChartSimple.swift:2]: I normally keep the headers consistent. The commits will still have the authors via PRs.

Comment (brennanMKE) [reply Swift Charts Examples/Charts/LineCharts/Simple/LineChartSimple.swift:2]: We could also maintain a CHANGELOG which can also link to PRs which is also good for visibility.

**Comment by brennanMKE** [comment]
> Also, I noticed that you left your development team in your commit.
> 
> Also, I think Swift_Charts_ExamplesApp is better than ChartsApp, since it's a clearer explanation of what the project is.

Ok. Just did not look right to me having underscores in the name.

**Comment by brennanMKE** [comment]
@jordibruin Just cleaned up the PR a bit. Had to update my Mac with some conveniences I had not put on here yet to help with staying in sync with the upstream. I pulled out the signing value for `DEVELOPMENT_TEAM` which only applies to me. I am not sure how to handle `SWIFT_VERSION` but it appears 5.7 is simply handled as Swift 5 by Xcode.

For releases what we could do is summarize all commits with each of the PRs which make up that release which would feature the developers who made contributions.

**Pull request closed without merging**